pkg/orchestrator/transformer/job.go

```diff
@@ -70,12 +69,14 @@ func applyBatchTaskDefaults(defaults types.BatchTaskDefaultConfig, task *models.
 		task.ResourcesConfig.GPU = defaults.Resources.GPU
 	}
 	// if the user didn't specify a publisher, but specified result path(s) assign the task a default publisher, if any.
-	if task.Publisher.IsEmpty() && len(task.ResultPaths) > 0 && defaults.Publisher.Config != "" {
-		config, err := job.ParsePublisherString(defaults.Publisher.Config)
-		if err != nil {
-			return fmt.Errorf("parsing default publisher spec (%s): %w", defaults.Publisher.Config, err)
-		}
-		task.Publisher = config
+	if task.Publisher.IsEmpty() && len(task.ResultPaths) > 0 && !defaults.Publisher.Config.IsEmpty() {
+		/*
+			config, err := job.ParsePublisherString(defaults.Publisher.Config)
+			if err != nil {
+				return fmt.Errorf("parsing default publisher spec (%s): %w", defaults.Publisher.Config, err)
+			}
+		*/
```

Comment:
commented out code.

Forrest, the current behaviour is we apply the default publisher regardless of whether a job has results paths or not. We don't need to change this behaviour, and obviously changing it is causing some other issues for you.

Remember you didn't like having a default publisher at all, and applying a publisher conditionally was the way you opted for to mitigate your concerns. As this PR is getting too big, lets avoid changing too many things in the same PR and lets just focus on just making the new config structure work

Reply:
okay will do